packages/unified-selection/src/unified-selection/iModel/IModel.ts

```diff
@@ -26,91 +24,93 @@ export enum SelectionSetEventType {
   Clear,
 }
 
+/**
+ * Payload sent to [[SelectionSet.onChanged]] event listeners to describe how the contents of the set have changed.
+ */
+export type SelectionSetEvent = SelectAddEvent | SelectRemoveEvent | SelectReplaceEvent;
+
+/**
+ * Interface representing a connection to iModel.
+ */
+export interface IModelSelection {
+  /** Key of the iModel. */
+  get key(): string;
+  /** The set of currently hilited elements for this IModelConnection. */
+  readonly hilited: IModelHiliteSet;
+  /** The set of currently selected elements for this IModelConnection. */
+  readonly selectionSet: SelectionSet;
+}
+
 /**
  * Passed to [[SelectionSet.onChanged]] event listeners when elements are added to the selection set.
  */
-export interface SelectAddEvent {
+interface SelectAddEvent {
   /** The type of operation that produced this event. */
   type: SelectionSetEventType.Add;
   /** The Ids of the elements added to the set. */
-  added: IdArg;
+  added: Id64Arg;
   /** The affected SelectionSet. */
   set: SelectionSet;
 }
 
 /**
  * Passed to [[SelectionSet.onChanged]] event listeners when elements are removed from the selection set.
  */
-export interface SelectRemoveEvent {
+interface SelectRemoveEvent {
   /** The type of operation that produced this event. */
   type: SelectionSetEventType.Remove | SelectionSetEventType.Clear;
   /** The element Ids removed from the set. */
-  removed: IdArg;
+  removed: Id64Arg;
   /** The affected SelectionSet. */
   set: SelectionSet;
 }
 
 /**
  * Passed to [[SelectionSet.onChanged]] event listeners when elements are simultaneously added to and removed from the selection set.
  */
-export interface SelectReplaceEvent {
+interface SelectReplaceEvent {
   /** The type of operation that produced this event. */
   type: SelectionSetEventType.Replace;
   /** The element Ids added to the set. */
-  added: IdArg;
+  added: Id64Arg;
   /** The element Ids removed from the set. */
-  removed: IdArg;
+  removed: Id64Arg;
   /** The affected SelectionSet. */
   set: SelectionSet;
 }
 
-/**
- * Payload sent to [[SelectionSet.onChanged]] event listeners to describe how the contents of the set have changed.
- */
-export type SelectionSetEvent = SelectAddEvent | SelectRemoveEvent | SelectReplaceEvent;
-
-/**
- * Manages a set of *listeners* for a particular event and notifies them when the event is raised.
- */
-export interface Event<T> {
-  /** Registers a Listener to be executed whenever this event is raised. */
-  addListener(listener: T, scope?: any): () => void;
-  /** Un-register a previously registered listener. */
-  removeListener(listener: T, scope?: any): boolean;
-}
-
 /**
  * A set of *currently selected* elements for an IModelConnection.
  */
-export interface SelectionSet {
+interface SelectionSet {
   /** iModel of the set. */
-  iModel: IModelConnection;
+  iModel: IModelSelection;
```

Comment:
It was used for checking whether the selection change event was of the correct iModel, but I removed it since the event iModel should always be the same.